src/components/Navbar.tsx

```diff
@@ -0,0 +1,183 @@
+          isSelected
+            ? style.primary
+            : theme?.colorMode === "dark"
+            ? style.lightBackground
+            : style.darkBackground
+        }
+      />
+      <div className={`navbar-item-title ${isSelected ? "selected" : ""}`}>
+        <span className="navbar-item-title-text">{title}</span>
+      </div>
+    </li>
+  );
+}
+
+type NavbarProps = {
+  label: string;
+  children: React.ReactNode;
+};
+
+export function Navbar(props: NavbarProps) {
+  const [selectedId, setSelectedId] = useState<number>();
+  const [items, setItems] = useState<React.RefObject<HTMLLIElement>[]>([]);
+  const selectedIndicatorRef = useRef<HTMLDivElement>(null);
+
+  // TODO: refactor this monstrosity (maybe ask Sandrina to give some tips)
+  const animateWithCss = (
+    domMenuItem: HTMLLIElement,
+    domSelectedIndicator: HTMLDivElement
+  ) => {
+    // First time an item is selected, the icon appears
+    if (!domSelectedIndicator.style.visibility) {
+      domSelectedIndicator.style.top = `${domMenuItem.offsetHeight - 12}px`;
+      domSelectedIndicator.style.left = `${
+        // center positioning the icon
+        domMenuItem.offsetLeft +
+        domMenuItem.clientWidth / 2 -
+        domSelectedIndicator.clientWidth / 2
+      }px`;
+      domSelectedIndicator.style.visibility = "visible";
+    } else {
+      // If an item is already selected, move the icon to next position
+      const currentPosition = domSelectedIndicator.offsetLeft;
+      const nextLeft = domMenuItem.offsetLeft;
+      const movement = nextLeft - currentPosition;
+      domSelectedIndicator.style.setProperty(
+        "--translate-x-value",
+        `${
+          movement +
+          domMenuItem.clientWidth / 2 -
+          domSelectedIndicator.clientWidth / 2
+        }px`
+      );
+      domSelectedIndicator.classList.add("animate");
+    }
+  };
+
+  useLayoutEffect(() => {
+    const domMenuItem = items[selectedId!]?.current;
+    const domSelectedIndicator = selectedIndicatorRef.current;
+    if (domMenuItem && domSelectedIndicator) {
+      animateWithCss(domMenuItem, domSelectedIndicator);
+    }
+  }, [items, selectedId]);
+
+  const registerItem = useCallback((item: React.RefObject<HTMLLIElement>) => {
+    setItems((prevItems) => [...prevItems, item]);
+  }, []);
+
+  return (
+    <ThemeProvider>
+      <NavbarContext.Provider
+        value={{
+          selectedId,
+          setSelectedId,
+          registerItem,
+        }}
+      >
+        <nav aria-label={props.label}>
+          <ul role="menubar" className="navbar">
+            {React.Children.map(props.children, (child, index) => {
+              if (isNavbarItem(child)) {
+                throw Error("Only NavbarItem allowed as child of Navbar");
+              }
+              return React.cloneElement(child as React.ReactElement, {
+                id: index,
+              });
+            })}
+            {selectedId !== undefined && (
+              <div
```

Comment:
bug: [ul > div is invalid html](https://caninclude.glitch.me/caninclude?child=div&parent=ul). this dot needs to be a child of nav.

Reply:
oh nice catch,
btw i love this site (didn't know it) and specifically this thing
<img width="134" alt="image" src="https://github.com/alexodan/navbar-challenge/assets/35075089/eff113fa-a7f1-4fc4-a8d4-2ec44ec29425">